pkg/controller/clusterserviceversion/clusterserviceversion_controller.go

```diff
@@ -196,5 +261,123 @@ func (r *ReconcileClusterServiceVersion) Reconcile(request reconcile.Request) (r
+	}
+
+	meterDefinition := &marketplacev1alpha1.MeterDefinition{}
+	_, err = meterDefinition.BuildMeterDefinitionFromString(meterDefinitionString, CSV.GetName(), CSV.GetNamespace(), utils.CSV_ANNOTATION_NAME, utils.CSV_ANNOTATION_NAMESPACE)
+	if err != nil {
+		reqLogger.Error(err, "Could not build a local copy of the MeterDefinition")
+		return err
+	}
+
+	err = r.client.Delete(context.TODO(), meterDefinition, client.PropagationPolicy(metav1.DeletePropagationForeground))
+	if err != nil && errors.IsNotFound(err) {
+		return err
+	}
+	reqLogger.Info("found and deleted MeterDefinition")
+	return nil
+
+}
+
+// reconcileMeterDefAnnotation checks the Annotations for the rhm CSV
+// If the CSV is new, we tag it and create a MeterDefinition
+// If the CSV is old, we check if the actual MeterDefinition matches the CSV (json) MeterDefinition
+func (r *ReconcileClusterServiceVersion) reconcileMeterDefAnnotation(CSV *olmv1alpha1.ClusterServiceVersion, annotations map[string]string) (reconcile.Result, bool, error) {
+	var err error
+	reqLogger := log.WithValues("CSV.Name", CSV.Name, "CSV.Namespace", CSV.Namespace)
+
+	// checks if it is possible to build MeterDefinition from annotations of CSV
+	reqLogger.Info("retrieving MeterDefinition string from csv")
+	meterDefinitionString, ok := annotations[utils.CSV_METERDEFINITION_ANNOTATION]
+	if !ok {
+		reqLogger.Info("No value for ", "key: ", utils.CSV_METERDEFINITION_ANNOTATION)
+		return reconcile.Result{}, false, nil
+	}
+
+	// builds a meterdefinition from our string (from the annotation)
+	reqLogger.Info("retrieval successful")
+	meterDefinition := &marketplacev1alpha1.MeterDefinition{}
+	_, err = meterDefinition.BuildMeterDefinitionFromString(meterDefinitionString, CSV.GetName(), CSV.GetNamespace(), utils.CSV_ANNOTATION_NAME, utils.CSV_ANNOTATION_NAMESPACE)
+	if err != nil {
+		reqLogger.Error(err, "Could not build a local copy of the MeterDefinition")
+		return reconcile.Result{}, true, err
+	}
+
+	//checks if the CSV is new or old (ie. is it already being tracked?)
+	reqLogger.Info("Checking if the csv is being tracked")
+	if val, ok := annotations[trackMeterTag]; ok && val == "true" {
+
+		// Case 1: The CSV is old: compare vs. expected MeterDefinition
+		reqLogger.Info("CSV is already tracked")
+
+		actualMeterDefinition := &marketplacev1alpha1.MeterDefinition{}
+		err = r.client.Get(context.TODO(), client.ObjectKey{Name: meterDefinition.GetName(), Namespace: meterDefinition.GetNamespace()}, actualMeterDefinition)
+		if err != nil {
+			reqLogger.Error(err, "Could not retrieve the existing MeterDefinition")
+			return reconcile.Result{}, true, err
+		}
+
+		if !reflect.DeepEqual(meterDefinition.Spec, actualMeterDefinition.Spec) && !reflect.DeepEqual(meterDefinition.ObjectMeta, actualMeterDefinition.ObjectMeta) {
+			err = emperr.New("Actual MeterDefinition and Expected MeterDefinition mismatch")
+			reqLogger.Error(err, "The actual meterdefinition is different from the expected meterdefinition")
```

Comment:
When they're not equal, we need to update the meterdefinition. You can create a patch to update with just the changes.